### PR TITLE
Support for Atmega 1284p and Wiznet W5200 / W5500 chips

### PR DESCRIPTION
**Pull request by arjenhiemstra**
Needed support for these chips for my own project. Now finally had the time to integrate back into the original code. Have a look and see if you want to merge it, it has been tested to work with Atmega 328p w/ W5100, w/ W5200 and w/ W5500 and Atmega 1284p w/ W5200 and w/ W5500.

### CODE REVIEW

**Comment by LoathingKernel** [comment]
Thank you for the PR. This has many changes and I am not familiar with W5200 or W5500 chips, so it will take me some time to test it but so far it looks really good and I definitely want to merge it.

**Comment by per1234** [comment]
I'm very excited to see this change made to Ariadne. Thanks!

I realize this would be a lot of work at this point but it would be really nice if these changes were broken into two pull requests: Adding ATmega1284P support, and Adding W5200/W5500 support. This would make for a much nicer commit history.

File: README.md [hunk 1]
```diff
@@ -343,7 +343,7 @@ Supported ethernet controllers:
 
 * __WizNet W5100__
 * __WizNet W5200__
-* __WizNet W5200__
+* __WizNet W5500__
```
Comment (per1234) [inline README.md]: I think this change should be squashed into https://github.com/codebendercc/Ariadne-Bootloader/pull/27/commits/f07bc14783323283061df9b32103aa46b2de3c3b.

File: hardware/ariadne/bootloaders/ariadne/w5100.h [hunk 2]
```diff
@@ -3,9 +3,9 @@
 
 
 /* Network settings */
-#define IP_ADDR     192,168,84,128
+#define IP_ADDR     192,168,1,128
 #define SUB_MASK    255,255,255,0
-#define GW_ADDR     192,168,84,1
+#define GW_ADDR     192,168,1,254
 #define MAC_ADDR    0xDE,0xAD,0xBE,0xEF,0xFE,0xED
```
Comment (per1234) [inline hardware/ariadne/bootloaders/ariadne/w5100.h]: I think this change should be squashed into https://github.com/codebendercc/Ariadne-Bootloader/pull/27/commits/e4d898b4e7d4fd62f79633d52c6442ea6c2ff69a

**Comment by arjenhiemstra** [comment]
Cool, thanks for this reply.
I can split the work into two PR's (the 1284p support is code wise not that heavy), let me see if I can find some time to do this somewhere this week.

**Comment by arjenhiemstra** [comment]
I will open a new PR with the proposed changes in two separate commits.

**Comment by LoathingKernel** [comment]
Thanks! Also, if you don't mind, don't include any compiled binaries in the PR, as the size of them makes diffs unavailable on github.

**Comment by arjenhiemstra** [comment]
Ok, let met find out how to do that (i’m not a git expert)

**Comment by LoathingKernel** [comment]
Nevermind, I will clean it up after merging it. I do not want to lose commit history so I will do some cleanups with later commit.

**Pull request closed without merging**